docs/madr/decisions/011-mesh-traffic-route.md

```diff
@@ -0,0 +1,119 @@
+# `MeshTrafficRoute`
+
+- Status: accepted
+
+Technical Story: https://github.com/kumahq/kuma/issues/4743
+
+## Context and Problem Statement
+
+Now that we have the [new policy matching resources](https://github.com/kumahq/kuma/blob/master/docs/madr/decisions/005-policy-matching.md), it's time to create one for traffic routung.
+We currently have both `TrafficRoute` and `MeshGatewayRoute`.
+
+This MADR does not address Gateway API integration.
+
+## Decision Drivers
+
+- `MeshGatewayRoute` and `TrafficRoute` have a large overlap in functionality
+  and the use cases for routing from a gateway and between services are similar
+- Gateway API has different resources per protocol
+
+## Considered Options
+
+- Two resources: `MeshGatewayRoute` and `MeshTrafficRoute`
+- Unified resource for services and gateway: `MeshTrafficRoute`
+- Unified resources but one per protocol, e.g. `MeshHTTPRoute`
+
+## Decision Outcome
+
+The new resource routes and alters requests depending on where it's coming from and where it's going to. Routes end up in the client-side Envoy config so we choose a `spec.to` resource.
+
+```yaml
+spec:
+ targetRef: ... # from where requests are coming
+ to:
+  - targetRef: ... # to where requests are going
+ default: ...
+```
+
+Each protocol gets its own resource:
+
+- `MeshHTTPRoute`
+- `MeshTCPRoute`
+
+which can be extended with others like GRPC.
```

Comment:
Might be explained later but what happens when all 3 are layered? The same dataplane has a TCP and HTTP and a GRPC route matching?

Reply:
IMO only whichever protocol is being used for the connection should be applied. Does that make sense?

Reply:
Right so the most specific? I think it makes sense but you should add a note abotu this here

Reply:
Added a note